koku/sources/koku_http_client.py

```diff
@@ -41,17 +43,91 @@ def __init__(self, auth_header):
         header = {'x-rh-identity': auth_header, 'sources-client': 'True'}
         self._identity_header = header
 
-    def create_provider(self, name, provider_type, authentication, billing_source):
-        """Koku HTTP call to create provider."""
-        url = '{}/{}/'.format(self._base_url, 'providers')
-        json_data = {'name': name, 'type': provider_type}
+    @staticmethod
+    def _get_dict_from_text_field(value):
+        try:
+            db_dict = json.loads(value)
+        except ValueError:
+            db_dict = {}
+        return db_dict
+
+    @staticmethod
+    def _build_provider_resource_name_auth(authentication):
+        if authentication.get('resource_name'):
+            auth = {'provider_resource_name': authentication.get('resource_name')}
+        else:
+            raise KokuHTTPClientError('Missing provider_resource_name')
+        return auth
+
+    @staticmethod
+    def _build_credentials_auth(authentication):
+        if authentication.get('credentials'):
+            auth = {'credentials': authentication.get('credentials')}
+        else:
+            raise KokuHTTPClientError('Missing credentials')
+        return auth
+
+    def _authentication_for_aws(self, authentication):
+        return self._build_provider_resource_name_auth(authentication)
+
+    def _authentication_for_ocp(self, authentication):
+        return self._build_provider_resource_name_auth(authentication)
```

Comment:
You're calling a staticmethod non-statically here. Perhaps the methods they're calling shouldn't have the `@staticmethod` decorator?

Reply:
Yeah, it does look odd I guess.  It needs a reference to itself to get the static method.

Reply:
To call it statically, you'd use this:

```
KokuHTTPClient._authentication_for_ocp()
```

But then lint would complain about you calling a class' private methods. :grin: 

So, as above, let's keep this code as-is, but drop the `@staticmethod` decorator from the method.

Reply:
Ok Ill remove `@staticmethod`.  My editor isn't happy with it but it seems to be getting past our tox/lint setup so lets do it.